### PR TITLE
Adds a cursed stomach as a new Bubblegum loot

### PR DESCRIPTION
**Pull request by ChesterTheCheesy**
# General Documentation
### Intent of your Pull Request
Adds a cursed demon stomach, that when consumed, gives the owner ability to vomitcrawl.
Stomach also allows person to vomit at will, to create pools of vomit to enter.
They only vomit if they're eligable for it (as in, races that don't hunger cannot vomit this way)

### Why is this change good for the game?
More loot that doesn't involve murderbone is good in my book.
Improves RP, obviously.
Marmio said that it also improves RP.

# Wiki Documentation
### Briefly describe your PR and the impacts of it, in layman's terms. 
Adds a new loot for killing bubblegum, very similar to demon heart, but instead of crawling in blood (which is everwhere),
you can crawl in vomit (which is nowhere, you'll have to vomit everywhere yourself). It is a neat item, but bear in mind you'll still have to kill the big daddy king of demons for it and hope for a 1/4 roll.

### What should players be aware of when it comes to the changes your PR is implementing?
new bubby loot baby!

### What general grouping does this PR fall under? 
Expanding mining by adding more unique loot

### If there are any numerical values involved in your PR that will be relevant to a player, please note them here. 
You can no longer vomit when crawling (take that, vomitghosts!)

# Changelog
:cl:  
rscadd: Added new component: vomitcrawl and an associated cursed stomach. Added it as new bubblegum loot.
bugfix: You can no longer vomit while in incorporeal form (this includes all forms of phasing).
imageadd: Added icon for cursed stomach .
/:cl:

https://user-images.githubusercontent.com/71487903/105613120-d430bd00-5dc0-11eb-82c4-9f6f3978f9b7.mp4

### CODE REVIEW

**Comment by UselessTheremin** [comment]
This is incredible

**Comment by patpol4** [comment]
EDIT: Ignore this im stupid
This increases the chances of rolling a crawl item from 33% to 50%
Vomit is rare though

**Comment by ChesterTheCheesy** [comment]
Erm, it's still a 1/4 loot, i didn't remove any other bubblegum loot from table.

File: code/modules/mob/living/carbon/carbon.dm [hunk 1]
```diff
@@ -438,6 +438,9 @@
 /mob/living/carbon/proc/vomit(lost_nutrition = 10, blood = FALSE, stun = TRUE, distance = 1, message = TRUE, toxic = FALSE)
 	if(HAS_TRAIT(src, TRAIT_NOHUNGER))
 		return 1
+		
+	if(istype(src.loc, /obj/effect/dummy))  //cannot vomit while phasing/vomitcrawling
+		return 1
```
Comment (redmoogle) [inline code/modules/mob/living/carbon/carbon.dm]:
```suggestion
		return TRUE
```

Code Standards. also please update the one above

File: code/modules/surgery/organs/stomach.dm [hunk 2]
```diff
@@ -125,3 +125,48 @@
 
 /obj/item/organ/stomach/ethereal/proc/adjust_charge(amount)
 	crystal_charge = clamp(crystal_charge + amount, ETHEREAL_CHARGE_NONE, ETHEREAL_CHARGE_FULL)
+	
+	/obj/item/organ/stomach/cursed
```
Comment (redmoogle) [inline code/modules/surgery/organs/stomach.dm]: Ident(_indent_)

**Review by redmoogle — DISMISSED** [review]
smh

File: code/modules/mob/living/carbon/carbon.dm (hunk 1 above)
Comment (ChesterTheCheesy) [reply code/modules/mob/living/carbon/carbon.dm]: Done

File: code/modules/surgery/organs/stomach.dm (hunk 2 above)
Comment (ChesterTheCheesy) [reply code/modules/surgery/organs/stomach.dm]: Done, thanks.

**Comment by Marmio64** [comment]
add a couple more 'improves RP' sentences in there so it has to be accepted despite the feature freeze. Also say it fixes a bug with what you added.

**Comment by LazennG** [comment]
pog, finally loot that isnt pretty much antag restricted and still has utility for nonantags aside from blood contract which is only good for seeing if someone is dead

**Comment by The-Sun-In-Splendour** [comment]
As a true RP-er I can assure you that this PR will raise the server's RP level so much the "RP" in the server name is gonna make much more sense

**Comment by Mesalikepie** [comment]
Does not raise RP but looks fun

**Review by JamieD1 — APPROVED** [review]
